test/e2e/deployment.go

```diff
@@ -1013,3 +1016,79 @@ func testScalePausedDeployment(f *framework.Framework) {
 		Expect(err).NotTo(HaveOccurred())
 	}
 }
+
+func testScaledRolloutDeployment(f *framework.Framework) {
```

Comment:
We should check that the strategy isn't violated (something like `framework.WaitForDeploymentStatus`) in this test.